src/objective-c/tests/UnitTests/APIv2Tests.m

```diff
@@ -770,4 +843,319 @@ - (void)testFlowControlReadNonBlockingFailure {
+
+  RMTStreamingOutputCallRequest *request = [RMTStreamingOutputCallRequest message];
+  RMTResponseParameters *parameters = [RMTResponseParameters message];
+  parameters.size = kSimpleDataLength;
+  [request.responseParametersArray addObject:parameters];
+  request.payload.body = [NSMutableData dataWithLength:kSimpleDataLength];
+
+  GRPCRequestOptions *requestOptions1 =
+      [[GRPCRequestOptions alloc] initWithHost:kHostAddress
+                                          path:kFullDuplexCallMethod.HTTPPath
+                                        safety:GRPCCallSafetyDefault];
+  GRPCRequestOptions *requestOptions2 =
+      [[GRPCRequestOptions alloc] initWithHost:kHostAddress
+                                          path:kFullDuplexCallMethod.HTTPPath
+                                        safety:GRPCCallSafetyDefault];
+
+  GRPCMutableCallOptions *callOptions = [[GRPCMutableCallOptions alloc] init];
+  callOptions.transportType = GRPCTransportTypeInsecure;
+  GRPCCall2 *call1 = [[GRPCCall2 alloc]
+      initWithRequestOptions:requestOptions1
+             responseHandler:[[ClientTestsBlockCallbacks alloc]
+                                 initWithInitialMetadataCallback:^(NSDictionary *initialMetadata) {
+                                   [metadata1 fulfill];
+                                 }
+                                 messageCallback:nil
+                                 closeCallback:^(NSDictionary *trailingMetadata, NSError *error) {
+                                   [completion1 fulfill];
+                                 }]
+                 callOptions:callOptions];
+  GRPCCall2 *call2 = [[GRPCCall2 alloc]
+      initWithRequestOptions:requestOptions2
+             responseHandler:[[ClientTestsBlockCallbacks alloc]
+                                 initWithInitialMetadataCallback:^(NSDictionary *initialMetadata) {
+                                   [metadata2 fulfill];
+                                 }
+                                 messageCallback:nil
+                                 closeCallback:^(NSDictionary *trailingMetadata, NSError *error) {
+                                   [completion2 fulfill];
+                                 }]
+                 callOptions:callOptions];
+  [call1 start];
+  [call2 start];
+  [call1 writeData:[request data]];
+  [call2 writeData:[request data]];
+  [self waitForExpectations:initialMetadataDone timeout:kTestTimeout];
+  GRPCCall2Internal *internalCall1 = call1->_firstInterceptor;
+  GRPCCall2Internal *internalCall2 = call2->_firstInterceptor;
+  XCTAssertEqual(
+      internalCall1->_call->_wrappedCall->_pooledChannel->_wrappedChannel->_unmanagedChannel,
+      internalCall2->_call->_wrappedCall->_pooledChannel->_wrappedChannel->_unmanagedChannel);
+  [call1 finish];
+  [call2 finish];
+  [self waitForExpectations:rpcDone timeout:kTestTimeout];
+}
+
+- (void)testChannelReuseDifferentCallSafety {
```

Comment:
Consider merging this and the previous test. You can change path and safety at the same time, and assert that the two channels are equal

Reply:
Removed the previous test case.

Reply:
That is the first half of the change, right? The second half is to change the `path` as well in this test, and verify that they still reuse the same channel.